src/Compilers/CSharp/Test/Syntax/Parsing/ForStatementParsingTest.cs

```diff
@@ -514,4 +514,508 @@ public void TestMultipleDeclaratorsWithInitializers()
+    public void TestMultipleDeclaratorsWithInitializers7()
+    {
+        UsingStatement("""
+            for (int offset = 0, c1 = , c2; offset < length;)
+            {
+            }
+            """,
+            // (1,27): error CS1525: Invalid expression term ','
+            // for (int offset = 0, c1 = , c2; offset < length;)
+            Diagnostic(ErrorCode.ERR_InvalidExprTerm, ",").WithArguments(",").WithLocation(1, 27));
+
+        N(SyntaxKind.ForStatement);
+        {
+            N(SyntaxKind.ForKeyword);
+            N(SyntaxKind.OpenParenToken);
+            N(SyntaxKind.VariableDeclaration);
+            {
+                N(SyntaxKind.PredefinedType);
+                {
+                    N(SyntaxKind.IntKeyword);
+                }
+                N(SyntaxKind.VariableDeclarator);
+                {
+                    N(SyntaxKind.IdentifierToken, "offset");
+                    N(SyntaxKind.EqualsValueClause);
+                    {
+                        N(SyntaxKind.EqualsToken);
+                        N(SyntaxKind.NumericLiteralExpression);
+                        {
+                            N(SyntaxKind.NumericLiteralToken, "0");
+                        }
+                    }
+                }
+                N(SyntaxKind.CommaToken);
+                N(SyntaxKind.VariableDeclarator);
+                {
+                    N(SyntaxKind.IdentifierToken, "c1");
+                    N(SyntaxKind.EqualsValueClause);
+                    {
+                        N(SyntaxKind.EqualsToken);
+                        M(SyntaxKind.IdentifierName);
+                        {
+                            M(SyntaxKind.IdentifierToken);
+                        }
+                    }
+                }
+                N(SyntaxKind.CommaToken);
+                N(SyntaxKind.VariableDeclarator);
+                {
+                    N(SyntaxKind.IdentifierToken, "c2");
+                }
+            }
+            N(SyntaxKind.SemicolonToken);
+            N(SyntaxKind.LessThanExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "offset");
+                }
+                N(SyntaxKind.LessThanToken);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "length");
+                }
+            }
+            N(SyntaxKind.SemicolonToken);
+            N(SyntaxKind.CloseParenToken);
+            N(SyntaxKind.Block);
+            {
+                N(SyntaxKind.OpenBraceToken);
+                N(SyntaxKind.CloseBraceToken);
+            }
+        }
+        EOF();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77160")]
+    public void TestMultipleDeclaratorsWithExpression1()
+    {
+        UsingStatement("""
+            for (Console.WriteLine("Blah"); true;)
```

Comment:
I was being serious about having more comprehensive testing of other possible forms. For example, let's look at something like: `for (MyType m = new() { A = 1,; true; m++)`. We need to have a _bunch_ of test additions here to make sure that we've adequately covered this testing gap.

Reply:
sgtm.

Reply:
added a lot more tets.